test/server/http/admin_test.cc

```diff
@@ -31,16 +34,21 @@ class AdminFilterTest : public testing::TestWithParam<Network::Address::IpVersio
 public:
   // TODO(mattklein123): Switch to mocks and do not bind to a real port.
   AdminFilterTest()
-      : admin_("/dev/null", TestEnvironment::temporaryPath("envoy.prof"),
+      : log_level_setter_(spdlog::level::warn),
```

Comment:
IIUC the setting here doesn't alter what the level of a generated message is.  It alters the severity threshold required for the log to pass through spdlog.  E.g. in the #2912 if I set the loglevel to 'error' rather than 'warn', it won't change the "mutation needs POST" message into an 'error' - that message is hard-coded to a warning, and it just won't show it, causing the test to fail.

I'll clarify in the comments in logging.h and logger.h

Reply:
Sorry that's not what I meant. What I meant is that if I want to run this test and see trace/debug/info trace messages, I can't. I have to go and change this code. This is one of the issues I was originally concerned about.

Reply:
I see -- you have a practice of running the test-binary with a command-line flag to override the loglevel to an even more permissive value than what I had set it to in this test, and you don't want to lose the ability to see all that when running tests manually.  Got it.

Maybe the model of what I'm trying to do needs to be adjusted so I don't mess that up.  You might actually want to set a lower permissiveness, or a higher permissiveness, than what's configured.  And what you really want is for the LogRecorder to catch all the messages, regardless of what the loglevel is set to, and for the loglevel to affect what's printed to the screen.

Maybe I need to put the record/delegate hooks at the Logger* level rather than the spdlog::sink level, so that it's loglevel independent.  WDTY?

Reply:
Yup exactly. Yeah I think we need to hook things in such a way that you can record logs independently somehow from the main logger, which is why I had suggested potentially allowing for multiple sinks, or multiple loggers, or something like that. (TBH I'm not really sure what the best way of accomplishing this is.)

Also, just to be clear, I think the delegate stuff is cool and makes the file logging cleaner, but I think we shouldn't lose the ability to dynamically change log levels during debugging.

Reply:
Do we need to catch all messages? It seems for most test purposes, you just need a narrow window around at most a few statements where you want to do modify the logging. See my other comment on `EXPECT_LOG`.